client.go

```diff
@@ -0,0 +1,104 @@
+package gotruevault
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const (
+	ContentTypeApplicationJSON = "application/json"
+)
+
+var (
+	// ErrUnauthorized ...
+	ErrUnauthorized = errors.New("error: authorization failed")
+
+	// ErrServerError ...
+	ErrServerError = errors.New("error: server error")
+
+	// ErrBadRequest ...
+	ErrBadRequest = errors.New("error: bad request")
+)
+
+//go:generate mockery --name URLBuilder
+type URLBuilder interface {
+	SearchDocumentURL(vaultID string) string
+}
+
+// defaultURLBuilder  ...
+type defaultURLBuilder struct{}
+
+// SearchDocumentURL ...
+func (t *defaultURLBuilder) SearchDocumentURL(vaultID string) string {
+	return fmt.Sprintf("https://api.truevault.com/v1/vaults/%s/search", vaultID)
+}
+
+type Client struct {
+	URLBuilder    URLBuilder
+	httpClient    *http.Client
+	authorization string
+}
+
+// New creates a TrueVault client
```

Comment:
The struct is `Client` if we keep saying `TrueVault Client` it might confuse people